File: Android task/MyApplication/app/src/main/java/com/example/myapplication/MainActivity.java
package com.example.myapplication;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import android.widget.TextView;
import android.widget.Button;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {
    EditText et1,et2;
    Button add,sub,mul,div;
    TextView tv;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        et1=findViewById(R.id.et1);
        et2=findViewById(R.id.et2);
        add=findViewById(R.id.add);
        sub=findViewById(R.id.sub);
        mul=findViewById(R.id.mul);
        div=findViewById(R.id.div);
        tv=findViewById(R.id.tv);
        add.setOnClickListener(new View.OnClickListener() {
                                   @Override
                                   public void onClick(View view) {
                                       int f,s,a;
                                       f=Integer.parseInt(et1.getText().toString());
                                       s=Integer.parseInt(et2.getText().toString());
                                       a=f+s;
                                       tv.setText("Result="+a);
                                   }
                               });

                sub.setOnClickListener(new View.OnClickListener() {
                                           @Override
                                           public void onClick(View view) {
                                               int f,s,a;
                                               f=Integer.parseInt(et1.getText().toString());
                                               s=Integer.parseInt(et2.getText().toString());
                                               a=f-s;
                                               tv.setText("Result="+a);
                                           }
                                       });

                        mul.setOnClickListener(new View.OnClickListener() {
                                                   @Override
                                                   public void onClick(View view) {
                                                       int f,s,a;
                                                       f=Integer.parseInt(et1.getText().toString());
                                                       s=Integer.parseInt(et2.getText().toString());
                                                       a=f*s;
                                                       tv.setText("Result="+a);
                                                   }
                                               });

                                div.setOnClickListener(new View.OnClickListener() {
                                                           @Override
                                                           public void onClick(View view) {
                                                               int f,s,a;
                                                               f=Integer.parseInt(et1.getText().toString());
                                                               s=Integer.parseInt(et2.getText().toString());
                                                               a=f/s;
                                                               tv.setText("Result="+a);
                                                           }
                                                       });

                ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
                    Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
                    v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
                    return insets;
                });
    }
}

File: LifecycleTask/app/src/main/java/com/example/lifecycletask/MainActivity.kt
package com.example.lifecycletask

import android.os.Bundle
import android.os.PersistableBundle
import android.util.Log
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import kotlin.math.log

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        Log.d("TAG", "onCreate: I am running")
        Toast.makeText(this,"on",Toast.LENGTH_SHORT).show()

        }
    

    override fun onStart() {
        super.onStart()
        Log.d("TAG", "onStart: I am running")
     Toast.makeText(this,"onStart",Toast.LENGTH_SHORT).show()
    }

    override fun onResume() {
        super.onResume()
        Log.d("TAG", "onStart: I am running")
        Toast.makeText(this,"onResume",Toast.LENGTH_SHORT).show()
    }

    override fun onPause() {
        super.onPause()
        Log.d("TAG", "onStart: I am running")
        Toast.makeText(this,"onPause",Toast.LENGTH_SHORT).show()
    }

    override fun onStop() {
        super.onStop()
        Log.d("TAG", "onStart: I am running")
        Toast.makeText(this,"onStop",Toast.LENGTH_SHORT).show()
    }

    override fun onRestart() {
        super.onRestart()
        Log.d("TAG", "onStart: I am running")
        Toast.makeText(this,"onRestart",Toast.LENGTH_SHORT).show()
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.d("TAG", "onStart: I am running")
        Toast.makeText(this,"onDestroy",Toast.LENGTH_SHORT).show()
    }

    }




